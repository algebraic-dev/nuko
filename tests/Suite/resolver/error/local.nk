
-- It works because it's not typechecking.
let print: Int = 42
let plus: Int  = 42

let local (a : Int): Int =
  let b = 3
  let c = plus a b
  if c
    then let e = 4
         plus e c
    else e