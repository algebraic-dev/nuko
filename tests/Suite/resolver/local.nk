
-- It works because it's not typechecking.
let print = 42
let plus  = 42

let local (a : Int) =
  let b = 3
  let c = plus a b
  if c
    then b
    else let d = 3
         plus d 4