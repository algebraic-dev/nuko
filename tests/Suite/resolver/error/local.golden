âœ— This
{"code":102,"place":"11:9-11:10","title":"Cannot find value from e"}
